### PR TITLE
test: add tests for update with exists=false precondition

### PR DESCRIPTION
**Pull request by dconeybe**
This is a follow-up to https://github.com/googleapis/conformance-tests/pull/82 which updated the test for the exists precondition.

### CODE REVIEW

**Comment by dconeybe** [comment]
@jskeet FYI

**Review by jskeet — APPROVED** [review]
Looks fine to me (albeit a little odd as this is effectively "create" isn't it?). Testing it against the current .NET implementation would be slightly fiddly, but I expect it to be fine.

File: firestore/v1/update-exists-false-precond.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      "description": "update: Exists=false precondition is valid",
+      "comment": "The Update method supports an explicit exists=false precondition.",
+      "update": {
+        "docRefPath": "projects/projectID/databases/(default)/documents/C/d",
+        "precondition": {
+          "exists": false
+        },
+        "jsonData": "{\"a\": 1}",
+        "isError": true,
+        "request": {
```
Comment (jskeet) [inline firestore/v1/update-exists-false-precond.json:13]: As noted in chat, let's remove this given that it's meant to be an error.

**Review by jskeet — CHANGES_REQUESTED** [review]
Undoing previous LGTM - it's not clear whether we *do* want to prohibit this. (We've never prohibited it in .NET - are we sure this won't break people? What does the server do?)

**Review by jskeet — APPROVED** [review]
Much clearer, thanks. And I've verified that it won't break the public API surface for .NET as we don't allow public specification of that precondition anywhere. (I'll need to add validation into the method in order to get the conformance tests to pass, but that's okay.)

**Comment by dconeybe** [comment]
@ddelgrosso1 Are you able and willing to give approval for this PR too? It's a follow-up to https://github.com/googleapis/conformance-tests/pull/82. Feel free to DM me and/or @jskeet if you would like any clarifications. Thank you.

**Pull request merged**